awswrangler/s3/_read_orc.py

```diff
@@ -0,0 +1,651 @@
+    _InternalReadTableMetadataReturnValue,
+    _TableMetadataReader,
+    _validate_schemas_from_files,
+)
+from awswrangler.typing import RaySettings, _ReadTableMetadataReturnValue
+
+if TYPE_CHECKING:
+    from mypy_boto3_s3 import S3Client
+
+FULL_READ_S3_BLOCK_SIZE = 20_971_520  # 20 MB (20 * 2**20)
+METADATA_READ_S3_BLOCK_SIZE = 131_072  # 128 KB (128 * 2**10)
+
+_logger: logging.Logger = logging.getLogger(__name__)
+
+
+def _pyarrow_orc_file_wrapper(source: Any) -> pyarrow.orc.ORCFile:
+    try:
+        return pyarrow.orc.ORCFile(source=source)
+    except pyarrow.ArrowInvalid as ex:
+        if str(ex) == "ORC file size is 0 bytes":
+            _logger.warning("Ignoring empty file...")
+            return None
+        raise
+
+
+@engine.dispatch_on_engine
+def _read_orc_metadata_file(
+    s3_client: Optional["S3Client"],
+    path: str,
+    s3_additional_kwargs: Optional[Dict[str, str]],
+    use_threads: Union[bool, int],
+    version_id: Optional[str] = None,
+) -> pa.schema:
+    with open_s3_object(
+        path=path,
+        mode="rb",
+        version_id=version_id,
+        use_threads=use_threads,
+        s3_client=s3_client,
+        s3_block_size=METADATA_READ_S3_BLOCK_SIZE,
+        s3_additional_kwargs=s3_additional_kwargs,
+    ) as f:
+        orc_file: Optional[pyarrow.orc.ORCFile] = _pyarrow_orc_file_wrapper(source=f)
+        if orc_file:
+            return orc_file.schema
+        return None
+
+
+def _read_schemas_from_files(
+    paths: List[str],
+    sampling: float,
+    use_threads: Union[bool, int],
+    s3_client: "S3Client",
+    s3_additional_kwargs: Optional[Dict[str, str]],
+    version_ids: Optional[Dict[str, str]] = None,
+) -> List[pa.schema]:
+    paths = _utils.list_sampling(lst=paths, sampling=sampling)
+
+    executor: _BaseExecutor = _get_executor(use_threads=use_threads)
+    schemas = ray_get(
+        executor.map(
+            _read_orc_metadata_file,
+            s3_client,
+            paths,
+            itertools.repeat(s3_additional_kwargs),
+            itertools.repeat(use_threads),
+            [version_ids.get(p) if isinstance(version_ids, dict) else None for p in paths],
+        )
+    )
+    return [schema for schema in schemas if schema is not None]
+
+
+class _ORCTableMetadataReader(_TableMetadataReader):
+    def read_schemas_from_files(
+        self,
+        paths: List[str],
+        sampling: float,
+        use_threads: Union[bool, int],
+        s3_client: "S3Client",
+        s3_additional_kwargs: Optional[Dict[str, str]],
+        version_ids: Optional[Dict[str, str]],
+        coerce_int96_timestamp_unit: Optional[str] = None,
+    ) -> List[pa.schema]:
+        return _read_schemas_from_files(
+            paths=paths,
+            sampling=sampling,
+            use_threads=use_threads,
+            s3_client=s3_client,
+            s3_additional_kwargs=s3_additional_kwargs,
+            version_ids=version_ids,
+        )
+
+
+def _read_orc_metadata(
```

Comment:
Is this wrapper needed? Reader looks good, would be great if we could use it directly in the upstream code